app/lib/history/models.dart

```diff
@@ -46,4 +52,61 @@ class History extends db.ExpandoModel {
   set paramsMap(Map<String, dynamic> value) {
     paramsJson = json.encode(value);
   }
+
+  HistoryEvent get historyEvent {
+    final fromJson = _events[type];
+    if (fromJson == null) return null;
+    return fromJson(paramsMap);
+  }
+
+  String formatMarkdown() => historyEvent?.formatMarkdown(this);
+}
+
+abstract class HistoryData {
+  String get packageName;
+  String get packageVersion;
+  DateTime get timestamp;
+}
+
+abstract class HistoryEvent {
+  String getType();
+
+  Map<String, dynamic> toJson();
+
+  String formatMarkdown(HistoryData data);
+}
+
+typedef HistoryEvent HistoryEventFromJson(Map<String, dynamic> json);
+
+final _events = <String, HistoryEventFromJson>{
```

Comment:
`_eventDeserializers`? I originally thought this held the actual events, which didn't really make sense to me. :)

Reply:
good point, fixed